### PR TITLE
SEP6-100: Add custom list + SEP6-105: Delete custom list

### PR DESCRIPTION
**Pull request by vladdirlea2002**
## How Has This Been Tested?

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your core changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: src/MovieManagement/Components/MovieLists/CreateMovieList.razor.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        try
+        {
+            await MovieListService.CreateCustomListAsync(_list);
+            _successMessage = "The new list has been successfully added";
+            await OnInitializedAsync();
+        }
+        catch (Exception e)
+        {
+            _errorMessage = e.Message;
+        }
```
Comment (RealFoXsTeRDOTA) [inline src/MovieManagement/Components/MovieLists/CreateMovieList.razor.cs:28]: May want to clear both successMessage and errorMessage before try/catch block.

File: src/MovieManagement/Components/MovieLists/DeleteMovieList.razor.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
+﻿namespace MovieManagement.Components.MovieLists;
+
+public partial class DeleteMovieList : ComponentBase
+{
+    [Parameter] 
+    public Guid ListId { get; set; }
+    private string _successMessage = default!;
```
Comment (RealFoXsTeRDOTA) [inline src/MovieManagement/Components/MovieLists/DeleteMovieList.razor.cs:7]: You could just explicitly set the message to an empty string either with "" or string.Empty in this scenario IMO

File: src/MovieManagement/Services/AzureService.cs [hunk 3]
```diff
@@ -15,6 +15,50 @@ public class AzureService : IAzureService
...
+        {
+            response.EnsureSuccessStatusCode();
+        }
+        catch
+        {
+            throw new Exception(response.Content.ReadAsStringAsync().Result);
+        }
+
+        return JsonSerializer.Deserialize<T>(await response.Content.ReadAsStringAsync(), _jsonSerializerOptions)!;
+    }
```
Comment (RealFoXsTeRDOTA) [inline src/MovieManagement/Services/AzureService.cs:39]: Could've probably just used the GetAsync and defaulted the body there to be null and instead passed in {id} with the endpoint but oh well.

File: src/MovieManagement/Services/MovieListService.cs [hunk 4]
```diff
@@ -0,0 +1,52 @@
+﻿namespace MovieManagement.Services;
+
+public class MovieListService : IMovieListService
+{
+    private static  List<MovieListViewModel> _movieLists = default!;
+    private readonly IAzureService _service;
+    private readonly AzureFunctionsConfig _settings;
+    public event EventHandler<EventArgs>? OnChanged;
```
Comment (RealFoXsTeRDOTA) [inline src/MovieManagement/Services/MovieListService.cs:8]: Any reason why you didn't just make an Action if you aren't using the parameters anyway?

File: src/MovieManagement/Shared/Sidebar.razor.cs [hunk 5]
```diff
@@ -2,23 +2,52 @@
...
+        await FetchMovieListsAsync();
+    }
+
+    private async Task FetchMovieListsAsync()
+    {
+        var userIdentity = (await AuthenticationStateProvider.GetAuthenticationStateAsync()).User.Identity;
+        if (userIdentity != null && userIdentity.IsAuthenticated)
+        {
+            var userId =
+                Guid.Parse((await AuthenticationStateProvider.GetAuthenticationStateAsync()).User.FindFirstValue("Id"));
```
Comment (RealFoXsTeRDOTA) [inline src/MovieManagement/Shared/Sidebar.razor.cs:34]: Extremely minor optimization (i know... - what else is there to review really 😄) but could probably just call the GetAuthenticationStateAsync once

File: src/MovieManagement/Shared/Sidebar.razor [hunk 6]
```diff
@@ -1,10 +1,14 @@
-﻿@inject NavigationManager NavigationManager
+﻿@using MovieManagement.Components.MovieLists
```
Comment (RealFoXsTeRDOTA) [inline src/MovieManagement/Shared/Sidebar.razor:1]: Oh god!! a @using statement.

... throw this in the _Imports file or whatever it's called

**Review by RealFoXsTeRDOTA — APPROVED** [review]

File: src/MovieManagement/Services/MovieListService.cs (hunk 4 above)
Comment (vladdirlea2002) [reply src/MovieManagement/Services/MovieListService.cs:8]: I don't know how =))

File: src/MovieManagement/Services/AzureService.cs (hunk 3 above)
Comment (vladdirlea2002) [reply src/MovieManagement/Services/AzureService.cs:39]: Fair. Just copy pasted the one from delete

**Pull request merged**